client/state/data-layer/wpcom/me/purchases/index.js

```diff
@@ -0,0 +1,61 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import makeJsonSchemaParser from 'lib/make-json-schema-parser';
+import schema from './schema';
+import {
+	PURCHASES_USER_FETCH,
+	PURCHASES_USER_FETCH_COMPLETED,
+	PURCHASES_USER_FETCH_FAILED,
+} from 'state/action-types';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {} from 'state/purchases/actions';
+
+/**
+ * Dispatches a request to fetch connected applications of the current user
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const fetchUserPurchases = action =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'GET',
+			path: '/me/purchases',
+		},
+		action
+	);
+
+/**
+ * @TODO (@sirreal)
+ *
+ * Replace purchases assembler invoked by selector with a transformer
+ */
+export const transformer = undefined;
+
+export default {
+	[ PURCHASES_USER_FETCH ]: [
+		dispatchRequestEx( {
+			fetch: fetchUserPurchases,
+			onSuccess: ( { userId }, purchases ) => ( {
+				type: PURCHASES_USER_FETCH_COMPLETED,
+				userId,
+				purchases,
+			} ),
+			onError: () => ( {
+				type: PURCHASES_USER_FETCH_FAILED,
+				error: translate( 'There was an error retrieving purchases.' ),
+			} ),
+			fromApi: makeJsonSchemaParser( schema, transformer ),
```

Comment:
note that instead of doing `integerish` in a JSON schema we can also just use the transformer directly with `parseInt( x, 10 )` or a variant that `throw`s when `NaN`

```js
const integer = v => {
  const n = parseInt( v, 10 );
  if ( isNaN( n ) ) {
    throw new ValueError();
  }
  return n;
}

const transformer = items => items.map( item => ( {
  id: integer( item.ID ),
  blogId: integer( item.blog_id ),
  userId: integer( item.user_id ),
  …
} )
```